File: tests/fixtures/sample-python-project/services/api_gateway/cache/redis_client.py
import os
import redis

redis_client = redis.Redis.from_url(os.environ.get("REDIS_URL", "redis://localhost:6379"))


def get_cache(key: str) -> str | None:
    return redis_client.get(key)


def set_cache(key: str, value: str, ttl: int) -> None:
    redis_client.set(key, value, ex=ttl)

File: tests/fixtures/sample-csharp-project-negative/services/ApiGateway/Cache/RedisClient.cs
using StackExchange.Redis;

namespace ApiGateway.Cache;

public static class RedisClient
{
    private static readonly ConnectionMultiplexer _redis = ConnectionMultiplexer.Connect(
        Environment.GetEnvironmentVariable("REDIS_URL") ?? "localhost:6379"
    );

    public static string? GetCache(string key)
    {
        var db = _redis.GetDatabase();
        return db.StringGet(key);
    }

    public static void SetCache(string key, string value, int ttl = 300)
    {
        var db = _redis.GetDatabase();
        db.StringSet(key, value, TimeSpan.FromSeconds(ttl));
    }
}
